src/Storage/Controllers/InstancesController.cs

```diff
@@ -118,13 +118,28 @@ public async Task<ActionResult<QueryResponse<Instance>>> GetInstances(InstanceQu
                 {
                     queryParameters.Org = queryParameters.AppId.Split('/')[0];
                 }
-
+                
                 if (!orgClaim.Equals(queryParameters.Org, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    return Forbid();
+                    if (string.IsNullOrEmpty(queryParameters.AppId))
+                    {
+                        return BadRequest("AppId must be defined.");
+                    }
+                    
+                    var appId = ValidateAppId(queryParameters.AppId).Split("/")[1];
+                    
+                    XacmlJsonRequestRoot request = DecisionHelper.CreateDecisionRequest(queryParameters.Org, appId, HttpContext.User, "read");
+                    XacmlJsonResponse response = await _authorizationService.GetDecisionForRequest(request);
+                    
+                    if (!DecisionHelper.ValidatePdpDecision(response?.Response, HttpContext.User))
+                    {
+                        return Forbid();
+                    }
+                }
+                else
+                {
+                    appOwnerRequestingInstances = true;
```

Comment:
I was thinking that we can allow this to be true for all application owners. Even when not the actual owner of the specific app.

Reply:
Alright!